samples/snippets/src/main/java/com/example/spanner/admin/generated/AddAndDropDatabaseRole.java

```diff
@@ -39,36 +40,41 @@ static void addAndDropDatabaseRole() throws IOException {
   }
 
   static void addAndDropDatabaseRole(
-      String projectId, String instanceId, String databaseId, String parentRole, String childRole)
-      throws IOException {
-    final DatabaseAdminClient databaseAdminClient = DatabaseAdminClient.create();
-    try {
-      System.out.println("Waiting for role create operation to complete...");
-      databaseAdminClient.updateDatabaseDdlAsync(
-              DatabaseName.of(projectId, instanceId, databaseId),
-              ImmutableList.of(
-                  String.format("CREATE ROLE %s", parentRole),
-                  String.format("GRANT SELECT ON TABLE Albums TO ROLE %s", parentRole),
-                  String.format("CREATE ROLE %s", childRole),
-                  String.format("GRANT ROLE %s TO ROLE %s", parentRole, childRole)))
-          .get(5, TimeUnit.MINUTES);
-      System.out.printf(
-          "Created roles %s and %s and granted privileges%n", parentRole, childRole);
-      // Delete role and membership.
-      System.out.println("Waiting for role revoke & drop operation to complete...");
-      databaseAdminClient.updateDatabaseDdlAsync(
-          DatabaseName.of(projectId, instanceId, databaseId),
-          ImmutableList.of(
-              String.format("REVOKE ROLE %s FROM ROLE %s", parentRole, childRole),
-              String.format("DROP ROLE %s", childRole))).get(5, TimeUnit.MINUTES);
-      System.out.printf("Revoked privileges and dropped role %s%n", childRole);
-    } catch (ExecutionException | TimeoutException e) {
-      System.out.printf(
-          "Error: AddAndDropDatabaseRole failed with error message %s\n", e.getMessage());
-      e.printStackTrace();
-    } catch (InterruptedException e) {
-      System.out.println(
-          "Error: Waiting for AddAndDropDatabaseRole operation to finish was interrupted");
+      String projectId, String instanceId, String databaseId,
+      String parentRole, String childRole) {
+    try (Spanner spanner =
```

Comment:
In these cases, you can combine these two `try-with-resources` blocks into one like this:

```java
try (Spanner spanner = SpannerOptions.newBuilder()
            .setProjectId(projectId)
            .build()
            .getService();
      DatabaseAdminClient databaseAdminClient = spanner.createDatabaseAdminClient()) {
   ....
}
```

That reduces the nested indentation in the sample.